File: arrays/merge_sorted_arrays/merge_sorted_arrays.ts
export class Solution {
  /**
   * @param {string} testInput
   * @return {any} Returns either boolean to validate the test cases or a custom result.
   */
  run_solution(testInput: any) {
    const [nums1, nums2, m, n] = testInput;

    let i = m - 1;
    let j = n - 1;
    let right = n + m - 1;

    while (j >= 0) {
      if (i >= 0 && nums1[i] < nums2[j]) {
        nums1[right] = nums2[j];
        j--;
      } else {
        nums1[right] = nums1[i];
        i--;
      }
      right--;
    }
    return nums1;
  }
}

File: arrays/merge_sorted_arrays/merge_sorted_arrays.py
"""Solution Module"""


class Solution:
    """Solution Class"""

    def run_solution(self, test_input):
        """Solution Method"""
        [nums1, nums2, m, n] = test_input
        i = m - 1
        j = n - 1
        right = (n + m) - 1

        while j >= 0:
            print(i, j, right, nums1)
            if i >= 0 and nums1[i] > nums2[j]:
                nums1[right] = nums1[i]
                i -= 1
            else:
                nums1[right] = nums2[j]
                j -= 1
            right -= 1

        return nums1
